### PR TITLE
describe the need for module-name

### PR DESCRIPTION
**Pull request by krpatter-intc**
This discussion was pretty critical for me:

https://github.com/PyO3/maturin/discussions/1540

And helped unblock some usages, so proposing that it get added to the readme here

### CODE REVIEW

**Comment by messense** [comment]
Thanks, but I dont think it's the right place to add it, better add it in user guide instead.

**Comment by krpatter-intc** [comment]
> Thanks, but I dont think it's the right place to add it, better add it in user guide instead.
> (We can keep the readme change)

Does that mean this PR is ok and I should also submit one for the user guide? or I should close/reject this PR as you dont want it here in the readme ?

**Comment by messense** [comment]
You can push new commits to add it to user guide, and we should mention that you need to also change your `#[pymodule]` name to `_lib_name`.

**Comment by krpatter-intc** [comment]
@messense updated with the need to specify the libname in the module

I'd like to do the User Guide as a separate PR. I think this is enough to help with some bare minimums. I'd like to spend a bit more time ramping on user guide layout in general. There's a few spots where I think an example or two would be helpful for newbies like myself.

File: README.md [hunk 1]
```diff
@@ -153,12 +154,24 @@ my-project
...
 │   ├── __init__.py
 │   ├── bar.py
-│   └── my_project.cpython-36m-x86_64-linux-gnu.so
+│   └── _lib_name.cpython-36m-x86_64-linux-gnu.so
 ├── README.md
 └── src
     └── lib.rs
 ```
 
+When doing this also be sure to set the module name in our code to match the Cargo.toml's `lib.name`
```
Comment (messense) [inline README.md:163]: It's actually the last part of `module-name` (split by `.`), `lib.name` in Cargo.toml doesn't matter when using `module-name`.

File: README.md [hunk 2]
```diff
@@ -153,12 +154,24 @@ my-project
...
 │   ├── __init__.py
 │   ├── bar.py
-│   └── my_project.cpython-36m-x86_64-linux-gnu.so
+│   └── _lib_name.cpython-36m-x86_64-linux-gnu.so
 ├── README.md
 └── src
     └── lib.rs
 ```
 
+When doing this also be sure to set the module name in your code to match the last part of module-name (don't include the package path):
```
Comment (messense) [inline README.md:163]:
```suggestion
When doing this also be sure to set the module name in your code to match the last part of `module-name` (don't include the package path):
```

**Review by messense — APPROVED** [review]

**Comment by krpatter-intc** [comment]
Thanks @messense !

**Pull request merged**